scalameta/dialects/shared/src/main/scala/scala/meta/Dialect.scala

```diff
@@ -158,7 +158,12 @@ final class Dialect private (
     // Scala 3 no longer allows def hello(){} - `=` is always needed
     val allowProcedureSyntax: Boolean,
     // Scala 3 no longer allows `do {...} while(...)`
-    val allowDoWhile: Boolean
+    val allowDoWhile: Boolean,
+    /* Kind-project support
+     * works under -Xsource3 flag
+     * https://github.com/scala/scala/pull/9605
+     */
+    val allowPlusMinusUnderscoreIdent: Boolean
```

Comment:
I would rename this. Looking at the scala code and doc, I'd call this `allowUnderscoreAsTypeWildcard` and set to false in scala3 and scala2source3. the plus/minus thing is just a side effect because it needs co/contravariant types.

Reply:
We already have `allowTypeParamUndesrcore` setting

Reply:
so why not use it? this is how it is described:
```
    // Dotty rejects placeholder as Type parameter
    val allowTypeParamUnderscore: Boolean,
```
yet it is never set to false anywhere and never used in parsing.

Reply:
I agree that this one fits better. 
But in this case non `Source3` scala2 dialects will accept this new syntax too.
Also, I forgot that we also need to enable `allowTypeParamUnderscore` for scala3

@kitbellew @tgodzik  are you ok with that?

Reply:
Oh. `allowTypeParamUnderscore` is about different thing.
```scala
def f[_](a: Int): Int = ???
```

Ok, then the question is: are you ok to enable this syntax by default without any flag?

Reply:
> Oh. `allowTypeParamUnderscore` is about different thing.
> 
> ```scala
> def f[_](a: Int): Int = ???
> ```

Right, it's for a type parameter of a definition, not for a type definition. still, why can't i find any reads for `allowTypeParamUnderscore`, and never `false` for it?

> Ok, then the question is: are you ok to enable this syntax by default without any flag?

`allowUnderscoreAsTypeWildcard` (or whatever you decide to call it) shouldn't be part of `scala213`.

i am thinking that we should implement some sort of reflection to allow a user to specify config
```
dialect {
  base = scala213
  options {
    allowUnderscoreAsTypeWildcard = true
  }
}
```
that way, we don't need to add it as part of any explicit dialect, we don't need `scala213source3` either.

Reply:
> Right, it's for a type parameter of a definition, not for a type definition. still, why can't i find any reads for allowTypeParamUnderscore, and never false for it?

It's used in several places. For example - [here](https://github.com/scalameta/scalameta/blob/55606a8a37c6be603c7eecc2639a75573af72ad7/scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala#L4114)

What about setting rename. I decided to keep the existing one.
I looked on it one more time and as we currently always parse `_` as `Type.Placeholder` without any settings so the name you proposed doesn't fit for that.

Reply:
> What about setting rename. I decided to keep the existing one.
> I looked on it one more time and as we currently always parse `_` as `Type.Placeholder` without any settings so the name you proposed doesn't fit for that.

Perhaps I should have been clearer. My proposal (with wildcards) would have had the opposite value; i.e., if you [still] allow wildcards, then it will not be parsed with +/-.